infra/gcp/roles/specs/iam.serviceAccountLister.yaml

```diff
@@ -0,0 +1,9 @@
+# formerly a project-level custom role for any project with GKE clusters,
+# not actually sure why this account is necessary but here we are
+title: Secret Account Lister
+description: Can list ServiceAccounts
+name: iam.serviceAccountLister
```

Comment:
This is is identical to the Role defined above in infra/gcp/roles/iam.serviceAccountLister.yaml  ??  Why do we have both?

Reply:
so I agree, it's silly in this case to have two files that look identical

but the idea is to consistently follow the pattern of: 
- custom roles are defined via a (custom) spec
- specs generate concrete roles
- custom role definitions are checked in to git (so we know _when_ custom roles get new permissions)
- then _those_ get applied

this lets us apply common defaults, and moves us toward a model of "write a spec, have something else apply it"

we can skip the "check in concrete roles" step and go straight to applying, but at the moment I find it useful to have them checked in for diffing when regeneration picks up new p

Reply:
So the non-spec is the readback?  I'm looking at the script and trying to figure out what it's doing.  It mixes the spec and the result of `gcloud iam roles describe`.  It's not clear to me when that script is run or who runs it or when.

Reply:
It is not currently run except by humans.  Now that @hh has proven pr-creator to be reusable out of test-infra, the last followup item I have for https://github.com/kubernetes/k8s.io/issues/1659 is to setup prowjobs that run this on postsubmit and periodically.

The reason for mixing the spec into comments of the generated role was to be 100% clear about what spec the role was generated from (vs. "this file", "this time", "this repo", "this hash" etc... but perhaps those would be a better / more concise approach to aim for)

Reply:
What I am missing is why we check the readback in, and how that is different from the audit.

IOW, why is it more than 
* Here's a spec
* Here's a script to apply the spec
* Here's a periodic script to audit the results
* Here's the last known-good audit state

If the periodic audit shows new perms, alert.  I'm missing the role of the intermediate?

I trust you so I am totally willing to OK this and just grow my comprehension async to this PR

Reply:
You may be right, this could be spurious and an unnecessary extra step.

I sorta want to keep this in as an experiment and remove later if we feel it's too annoying.  Reasons why:
- this might make it easier to vet audit output "here's what I absolutely expect it to be", and regenerating from primitive roles at time of audit might pick up new permissions I didn't know about last time I said "sure, refresh the role"
- it may be easier for contributors to find out "ok but what _exact_ permissions does this role have" even if they don't have iam permissions or gcloud handy

I guess I sort of think of this as the equivalent to a terraform state file.  What was the state of the world the last time I managed it.  Compare that to what the state of the world is now (did something else change things?) + the changes I intend to make to the state of the world.